Feature/caching

camilo960321 opened the pull request:
Testing for caching and and caching in EmployeeController.

camilo960321 committed:
cache xml

camilo960321 committed:
enable caching

camilo960321 committed:
Cache implementation

camilo960321 committed:
Correction of some errors

Commit:
Fixing some errors in cache implementation

Commit:
Fixing request mapping for caching

Commit:
Testing caching

camilo960321 committed:
Correction of some errors

camilo960321 committed:
Final PR for testing caching.

Gasiyu commented on src/main/kotlin/org/hyperskill/phonebook/controller/AuthController.kt:
```diff
@@ -14,6 +16,7 @@ import org.hyperskill.phonebook.dtos.Request
 class AuthController(
     private val authService: AuthService,
     private val jwtService: JwtService,
+    private val userPhonebookService: UserPhonebookService
```
Looks like `UserPhonebookService` service is unused in the `AuthController`, so it can be removed

Gasiyu commented on src/main/kotlin/org/hyperskill/phonebook/controller/EmployeeController.kt:
The `@Cacheable` annotation and `@CacheEvict` should be on the `EmployeeService` instead of `EmployeeController`, and it should annotate the method, not a controller

Gasiyu reviewed (changes requested):
The `@Cacheable` annotation and `@CacheEvict` should be on the `Service` instead of `Controller`, and it should annotate the method, not a controller #CMIIW

camilo960321 committed:
Putting caching in services instead of controllers.

camilo960321 committed:
Deleting caching controller

camilo960321 committed:
Changes in testing for cachePut

Gasiyu committed:
refactor(controller): remove unused imports and dependency from AuthController

- eliminate unnecessary imports to clean up the code
- remove userPhonebookService dependency for improved class focus

Gasiyu committed:
Merge remote-tracking branch 'origin/main' into feature/caching

Gasiyu committed:
feat(caching): revamp caching implementation and tests

- replace old cachePhoneBook with employees and users caches
- introduce Caching annotations for EmployeeService and UserService
- add extensive caching tests for employees and users
- update schema and ehcache.xml with new cache configurations
- include mockito-kotlin for caching-related tests in build dependencies

Gasiyu committed:
feat(caching): enhance caching behavior and update tests

- add @Caching annotations with evict rules for store and update operations in EmployeeService
- update EmployeeServiceCachingTest to validate cache eviction and replacement scenarios
- introduce getEmployee endpoint in EmployeeController for detailed employee retrieval with role-based DTOs

Gasiyu committed:
feat(caching): enhance caching eviction logic and update tests

- modify cache eviction keys in EmployeeService to support page-specific caching
- update EmployeeServiceCachingTest to verify eviction for all-page and filtered-page caches

Gasiyu reviewed (approved):
I added some test cases and moved the caching annotation into the existing method. Thanks for the work, @camilo960321! I'll merge this PR early, just in case Jose and the team are checking the answer
